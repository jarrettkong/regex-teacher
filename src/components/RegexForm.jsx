import React, { Component } from 'react';
import './RegexForm.scss'

class InputForm extends Component {
  
  state = {
    inputValue: ''
  }

  _handleChange = e => {
    this.setState({ inputValue: e.target.value }, console.log(this.state.inputValue));
  }

  render() {
    return (
      <header className="RegexForm-input">
        <label htmlFor="regex-input">Regular Expression</label>
        <input type="text" id="regex-input" placeholder="Type your regular expression here..." value={this.state.inputValue} onChange={this._handleChange}/>
      </header>
    )
  }
}

export default InputForm
