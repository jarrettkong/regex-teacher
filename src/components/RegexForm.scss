.RegexForm {
  width: 100%;
  display: flex;
  
  & input {
    width: 100%;
    height: 40px;
    padding-left: 15px;
    font-size: 17px;
    border: 0;
    border-radius: 2px;
    margin-top: 5px;
    color: #DE6A73;
    font-family: 'Roboto Mono', monospace;

    &:focus {
      outline: transparent;
    }
  }
}

.main-container {
  flex: 8;
  margin-right: 30px;
}

.flag-container {
  flex: 2;
}

.RegexForm-submit {
  display: none;
}