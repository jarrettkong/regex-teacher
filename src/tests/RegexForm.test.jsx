import React from 'react';
import RegexForm from '../components/RegexForm.jsx'
import { shallow } from 'enzyme';

const mockCheckAnswer = jest.fn();

describe('RegexForm', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <RegexForm checkAnswer={mockCheckAnswer} />
    )
  });

  it('should render the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have proper default state', () => {
    expect(wrapper.state()).toEqual({
      pattern: '',
      flags: ''
    });
  })

  describe('_handlePattern', () => {

    it('should update state on call', () => {
      expect(wrapper.state('pattern')).toEqual('');
      wrapper.find('#pattern-input').simulate('change', { target: { value: 'pattern' } })
      expect(wrapper.state('pattern')).toEqual('pattern');
    })

  })

  describe('_handleFlags', () => {

    it('should update state on call', () => {
      expect(wrapper.state('flags')).toEqual('');
      wrapper.find('#flag-input').simulate('change', { target: { value: 'flags' } })
      expect(wrapper.state('flags')).toEqual('flags');
    })

  })

  describe('_handleSubmit', () => {

    it('should call checkAnswer on submit', () => {
      wrapper.find('.RegexForm').simulate('submit', { preventDefault: () => {} });
      expect(mockCheckAnswer).toBeCalled();
    });
  });
  

});

