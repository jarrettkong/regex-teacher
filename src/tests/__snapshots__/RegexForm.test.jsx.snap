// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegexForm should render the snapshot with all the data passed in 1`] = `
<form
  className="RegexForm"
  onSubmit={[Function]}
>
  <div
    className="Regex-input-container main-container"
  >
    <label
      htmlFor="pattern-input"
    >
      Pattern
    </label>
    <input
      id="pattern-input"
      minLength="1"
      onChange={[Function]}
      placeholder="/pattern/"
      spellCheck="false"
      type="text"
      value=""
    />
  </div>
  <div
    className="Regex-input-container flag-container"
  >
    <label
      htmlFor="flag-input"
    >
      Flags
    </label>
    <input
      id="flag-input"
      onChange={[Function]}
      pattern="[gimsxeU]*"
      placeholder="/flags"
      spellCheck="false"
      type="text"
      value=""
    />
  </div>
  <input
    className="RegexForm-submit"
    type="submit"
  />
</form>
`;
